### ISSUE TITLE
Enhance components of rulebased with screen reader attributes

### ISSUE THREAD

**Issue by Larisa-Staroverova** [SAP/spartacus#14156]
This task could be split dependent on involved developers
Verify whether the screen reader reads everything as expected
Clarify what and how should be read with PO / UX
Enhance /add `aria` accordingly
